components/esp32/panic.c

```diff
@@ -228,6 +229,32 @@ static void disableAllWdts() {
 
 #endif
 
+static inline bool stackPointerIsSane(uint32_t sp) {
+	return !(sp < 0x3ffae010 || sp > 0x3ffffff0 || ((sp & 0xf) != 0));
+}
+static void putEntry(uint32_t pc, uint32_t sp) {
+	char buf[24];
+	if (pc & 0x80000000) pc = (pc & 0x3fffffff) | 0x40000000;
+	sprintf(buf, " %p:%p", (void *) pc, (void *) sp);
```

Comment:
Could you please modify this to use the panicPutHex function? The rest of the panic functionality has been written to avoid newlib, it would be nice to keep the same here.